### PR TITLE
Revamps gun damage modifiers, part one

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Balance "Readjusts the damage modifiers for the.223, .308, .50, and.30-06 caliber guns to bring them in line with the currently outlined methods for calculating such numbers."

#### Purpose of change
Over the past experimental cycle, the method with which a gun’s damage modifier based upon its barrel length was calculated has been drastically changed from the old system. Previously, the game balance file dictated that each caliber had a base barrel length and that an addition or subtraction of one damage was to be granted for each 3-inch deviation that a weapon’s barrel possessed from the base, defined barrel length. A.223-caliber weapon with a 19-inch barrel, for example, increases its overall damage by one point if the listed base barrel length is 16 inches.
Simple and long in use, this system has recently been deprecated in favor of a newer one that requires a formula to be applied to each weapon. Given its newer status and the multitude of weapons currently in the game, most of the in-game guns have not been updated to reflect the new system. In addition, the game balance file, which lists each calibre’s base barrel length, damage, and other needed information, also outlines cartridge damages that are slightly different from those implemented in the game.

#### Describe the solution
Damage amounts for.50,.223,.308, and.30-06 caliber cartridges have been updated to correspond with those in the game balance file. I have also calculated new damage modifiers for all the guns that fall under the aforementioned calibres, utilizing the formula outlined in the same document. To this end, I separated the Mk 18 CQBR from the M4A1, as its barrel length differed sufficiently from a standard M4 to necessitate a different damage modifier.

#### Describe alternatives you've considered
I contemplated working on updating common pistol calibres, but I felt that rifle rounds were more susceptible to notable changes under the new system than handguns and, as such, were more worthy of being cleared out.

#### Testing
I have not tested these changes as of yet, both due to a lack of personal time to run such tests as well as the fact that I wanted to ensure that I was calculating the modifiers correctly.

#### Additional context
For ease of reference, I have compiled a spreadsheet containing all of the weapons I have adjusted, the formulas I have used, and the calculated damage modifiers. Each listed barrel length has been sourced from Wikipedia and online manufacturer data, where applicable.
[Firearm Damage Modifiers.xlsx](https://github.com/CleverRaven/Cataclysm-DDA/files/10391284/Firearm.Damage.Modifiers.xlsx)

### CODE REVIEW

File: data/json/items/gun/223.json [hunk 1]
```diff
@@ -432,6 +429,53 @@
       }
     ]
   },
+  {
+    "id": "m4_cqbr",
+    "copy-from": "m4a1",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "special duty carbine" },
+    "description": "a rugged military carbine that has been outfitted with various accessories, such as an adjustable buttstock, suppressor, and shortened barrel, to improve its close-quarter battle effectiveness.  Most commonly deployed with special forces units and marines before the Cataclysm, its handy size lends itself well to room clearing and fighting within tight spaces.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/223.json:438]:
```suggestion
    "description": "A rugged military carbine that has been outfitted with various accessories, such as an adjustable buttstock, suppressor, and shortened barrel, to improve its close-quarter battle effectiveness.  Most commonly deployed with special forces units and marines before the Cataclysm, its handy size lends itself well to room clearing and fighting within tight spaces.",
```

**Comment by RenechCDDA** [comment]
The value for AR-10 is erroneous (should be ~58.5666, rounded to 59; not 56).

**Comment by anoobindisguise** [comment]
So in terms of game balance from the PR thus far:
- .30-06 guns lost about 5 damage (62 + 0 -> 49+8)
- .50 BMG gained 10 damage when used with Browning (131 + 0 -> 142 - 1) and lost 5 damage (131 - 5 -> 142 - 21) with other guns like Barrett.
- 7.62x51 are similar to before (bullet went from 60 to 54 but many guns got similar damage boosts to compensate) although some guns such as the SCAR-H lost about 6 damage.
- 223 are similar though some guns got worse (handguns and mini-14 lost 4-5 damage each for example)

So in summary a lot of nerfs, were these calibers overperforming compared to our damage estimation system?

**Comment by DoctorBoomstick** [comment]
I suppose how the new damages effect the overall flow and balance of gameplay is one of those matters that have yet to be field tested. I suppose that the small (3 or 4) damage deviations shouldn’t cause a drastic or notable change, though they might be a bit more apparent in the bigger calibres that have taken a hit. However, if the downward trend continues, smaller calibres such as.22 LR may become obsolete for basic zombie killing.
Speaking as someone not too familiar with the lore, I'd guess that these nerfs correspond to how effectively undead or non-biology-conforming entities that can survive having their faces pulverized would react to being shot. A hulk or other highly evolved zombie will be able to withstand gunfire more effectively than regular people, and zombies in general will be less concerned with bits of themselves being blown off. In general, I do personally feel like damage was just a smidge overstated, especially when you started utilizing guns on larger, more heavily evolved enemies.

**Comment by LyleSY** [comment]
Something's gone wrong here

**Comment by DoctorBoomstick** [comment]
> Something's gone wrong here

Indeed, and I am ignorant as to what and how to fix it. I just tried to rebase to latest experimental, and ah… this nonsense happened.

**Comment by DoctorBoomstick** [comment]
Some very confusing merge/ rebase related issues took place here, the solutions to which I am unsure of. Plus, it looks like the checks are failing as the damage modifiers for the .50 caliber guns exceed the supported possible ranges.
([slow] ~starting_items)=> value for damage outside supported range

    "ammo": [ "50" ],
    "range": 100,
    "ranged_damage":
                    ^
                     { "damage_type": "bullet", "amount": -21 },
    "dispersion": 130,
    "durability": 8,
Short of outright removing these modifiers, there doesn’t seem to be a way of fixing the issue. The game itself also presents this error when trying to utilise the modifiers in question. Given these problems, the latter of which completely negates the point for this PR’s existence, I’m taking it down for the time being until I can put it back up with the issues fixed.
